internal/scanner/scanner_test.go

```diff
@@ -1,43 +1,54 @@
 package scanner
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
+	"net/http/httptest"
 	"net/url"
 	"testing"
 
+	"github.com/kubewarden/audit-scanner/internal/constants"
+	"github.com/kubewarden/audit-scanner/internal/report"
+	"github.com/kubewarden/audit-scanner/internal/resources"
+	policiesv1 "github.com/kubewarden/kubewarden-controller/pkg/apis/policies/v1"
 	admv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-const auditURL = "http://localhost:9876/audit/"
-
-func startServer() {
-	mux := http.NewServeMux()
-	mux.HandleFunc("/audit/", func(response http.ResponseWriter, req *http.Request) {
-		_, err := io.Copy(response, req.Body)
-		if err != nil {
-			fmt.Fprintf(response, "Cannot write the response")
-		}
-	})
-	mux.HandleFunc("/", func(response http.ResponseWriter, req *http.Request) {
-		// The "/" pattern matches everything, so we need to check
-		// that we're at the root here.
-		if req.URL.Path != "/" {
-			http.NotFound(response, req)
-			return
-		}
-		fmt.Fprintf(response, "Welcome to the home page!")
-	})
-	err := http.ListenAndServe("localhost:9876", mux) //nolint
+var defaultHTTPTestingServerFunc = func(response http.ResponseWriter, req *http.Request) {
+	_, err := io.Copy(response, req.Body)
 	if err != nil {
-		fmt.Println(err)
+		fmt.Fprintf(response, "Cannot write the response")
 	}
 }
 
+var defaultHTTPTestingServerHandler = http.HandlerFunc(defaultHTTPTestingServerFunc)
+
+type ResourcesFetcherMock struct {
+	auditURL string
+}
+
+func (r ResourcesFetcherMock) GetResourcesForPolicies(_ context.Context, _ []policiesv1.Policy, _ string) ([]resources.AuditableResources, error) {
```

Comment:
I think that by having an in-mem store, that can be better mocked, we may not need this mocks.